auslib/db.py

```diff
@@ -1663,15 +1688,17 @@ def getRawMatches():
             if not self._buildIDMatchesRule(rule['buildID'], updateQuery['buildID']):
                 self.log.debug("%s doesn't match %s", rule['buildID'], updateQuery['buildID'])
                 continue
+            if not self._memoryMatchesRule(rule['memory'], updateQuery.get("memory", "")):
+                self.log.debug("%s doesn't match %s", rule['memory'], updateQuery.get("memory"))
+                continue
             # To help keep the rules table compact, multiple OS versions may be
             # specified in a single rule. They are comma delimited, so we need to
             # break them out and create clauses for each one.
             if not self._simpleBooleanMatchesRule(rule['osVersion'], updateQuery['osVersion']):
                 self.log.debug("%s doesn't match %s", rule['osVersion'], updateQuery['osVersion'])
                 continue
-            # Same deal for system capabilities
-            if not self._simpleBooleanMatchesRule(rule['systemCapabilities'], updateQuery.get('systemCapabilities', ""), substring=False):
-                self.log.debug("%s doesn't match %s", rule['systemCapabilities'], updateQuery.get('systemCapabilities'))
+            if not self._csvMatchesRule(rule['instructionSet'], updateQuery.get('instructionSet', ""), substring=False):
```

Comment:
I had one more thought. During deployment, what happens when the db is migrated and a canary public node is up, but the rest of the public nodes on the old code ? Perhaps the latter will hit a lot of errors when rule['systemCapabilities'] doesn't exist.